metabase/metabase
# Toggle series visibility from chart legend (cartesian charts)

Implements #2697 for cartesian charts ([product doc](https://www.notion.so/metabase/Let-users-hide-show-series-by-clicking-on-the-legend-90a5ec138b534f039136ea7f377252b8?pvs=4))

Allows to toggle series visibility from the chart legend by clicking the legend item dot (FWIW, series visibility set like that isn't meant to be persisted). Also, updates the `graph.series_order` visualization setting widget to better distinguish its "hide" action from the new legend behavior.

Currently supported charts: bar, line, area, combo, scatter (going to put together a separate PR for pie charts)

### To verify

1. Create a multi-series chart (e.g., Count of Orders per Product Category)
2. Ensure you can click the legend dots to show/hide the series
3. Ensure you can't hide the last series
4. In chill mode, ensure you can manage series order and related settings via the "Data" tab

### Demo

https://github.com/user-attachments/assets/a2c82570-6cb2-4825-a70e-a0a023a61305
